integration-tests/awsguard.go

```diff
@@ -0,0 +1,137 @@
+	case "", "advisory", "mandatory", "disabled":
+		// OK
+		break
+	default:
+		return errors.Errorf("unrecognized default enforcement level %q", settings.defaultEnforcementLevel)
+	}
+
+	// disabled rules
+	for _, rule := range settings.disableRules {
+		if !ruleNameRE.MatchString(rule) {
+			return errors.Errorf("rule name %q appears to be invalid", rule)
+		}
+	}
+
+	return nil
+}
+
+// CreatePolicyPack creates a new Node module in a sub folder of the test environment.
+// The awsGuardSettings will be written into the module's index.ts file.
+// Returns the path to the created module's directory.
+func (settings awsGuardSettings) CreatePolicyPack(e *ptesting.Environment) (string, error) {
+	e.Log("Creating customized AWS Guard module")
+
+	if err := settings.validate(); err != nil {
+		return "", errors.Wrap(err, "validation error")
+	}
+
+	initialCWD := e.CWD
+
+	moduleFolder := path.Join(e.RootPath, "custom-awsguard")
+	if err := os.Mkdir(moduleFolder, os.ModeDir|os.ModePerm); err != nil {
+		return "", errors.Wrap(err, "creating folder for customized AWS Guard module")
+	}
+
+	// package.json, defining the module itself.
+	packageJSONFilePath := path.Join(moduleFolder, "package.json")
+	packageJSONFileContents := `{
+		"name": "custom-awsguard",
+		"version": "1.0.0",
+		"description": "Customized AWS Guard policy pack for integration tests.",
+		"main": "index.js",
+		"dependencies": {
+		  "@pulumi/awsguard": "latest"
+		}
+	  }`
+	if err := ioutil.WriteFile(packageJSONFilePath, []byte(packageJSONFileContents), os.ModePerm); err != nil {
+		return "", errors.Wrap(err, "writing package.json")
+	}
+
+	// index.js, which contains encodes the settings via code.
+	indexTsFilePath := path.Join(moduleFolder, "index.js")
+	indexTsFileContents := settings.renderIndexJSFile()
+	if err := ioutil.WriteFile(indexTsFilePath, []byte(indexTsFileContents), os.ModePerm); err != nil {
+		return "", errors.Wrap(err, "writing index.js")
+	}
+
+	// Run `yarn` in the custom policy packs folder, to download the AWS Guard module as
+	// a dependency. (Which should have been "linked" with the development version in this
+	// repository.)
+	e.CWD = moduleFolder
+	e.RunCommand("yarn")
+	e.CWD = initialCWD
+
+	return moduleFolder, nil
+}
+
+// renderIndexJSFile returns the contents of the customized index.js file.
+func (settings awsGuardSettings) renderIndexJSFile() string {
+	contents := bytes.NewBufferString(`// Generated code. Do not edit.
+let awsguard = require("@pulumi/awsguard");
+module.exports.policyPack = new awsguard.AwsGuard({
+`)
+
+	// Write the default enforcement level.
+	if settings.defaultEnforcementLevel == "" {
+		settings.defaultEnforcementLevel = "mandatory"
+	}
+	contents.WriteString(fmt.Sprintf("\t'all': '%s',\n", settings.defaultEnforcementLevel))
+
+	// Configure every policy we wish to disable.
+	for _, ruleToDisable := range settings.disableRules {
+		line := fmt.Sprintf("'%s': { 'enforcementLevel': 'disable' },", ruleToDisable)
```

Comment:
This isn't going to work consistently for all policies, as not all policies accept their own bag of args -- policies without args can only be set to the enforcement level (which will also work for policies with args).

Also the string should be `disabled` (trailing 'd'), not `disable`.

```suggestion
		line := fmt.Sprintf("'%s': 'disabled',", ruleToDisable)
```

Reply:
Aside: Any reason not to use TypeScript for the generated file? Wondering if our built-in `ts-node` support would have caught the `disabled` vs. `disable` issue.